core/src/main/java/org/apache/calcite/util/format/FormatElementEnum.java

```diff
@@ -132,17 +133,41 @@ public enum FormatElementEnum implements FormatElement {
       sb.append(String.format(Locale.ROOT, "%02d", calendar.get(Calendar.HOUR_OF_DAY)));
     }
   },
+  ID("u", "The weekday (Monday as the first day of the week) as a decimal number (1-7)") {
+    @Override public void format(StringBuilder sb, Date date) {
+      final Calendar calendar = Work.get().calendar;
+      calendar.setTime(date);
+      int weekDay = calendar.get(Calendar.DAY_OF_WEEK);
+      sb.append(String.format(Locale.ROOT, "%d", weekDay == 1 ? 7 : weekDay - 1));
```

Comment:
Does this just convert from Sunday being 1 to Monday being 1?

Reply:
Yes. Maybe change the comparison to `weekDay == Calendar.SUNDAY` to make it clearer? Do you see another way?

Reply:
That could work, a simple comment would help also. It is no big deal either way :)

Reply:
In my mind Calcite is complex enough that when we have opportunities to explain little pieces of logic to reduce mental load for the developer, we should

Reply:
The DOW function implementation assumes that Sunday = 1.
In Postgres DOW assumes that Sunday = 0.

Reply:
Added a comment - don't think my suggestion would make it clearer.

Looking at Postgres documentation, the `EXTRACT(DOW...` assumes Sunday = 0, but the format functions `to_char, to_date,...` also assume Sunday = 1 for day of week.